### ISSUE TITLE
index.update does not work

### ISSUE THREAD

**Issue by NCKLS** [pinecone-io/pinecone-ts-client#38]
```
const update = {
    id: embedding,
    namespace: 'my-namespace',
    setMetadata: {
        metadata: myData,
    },
}
await index.update({ updateRequest: update })
```

I am able to copy and paste the values of the update object and see the results via the pinecone API example tools, but using the pinecone-ts-client I receive no object,  no error, and the embedding is not updated inside of the index.

It appears in this function: https://github.com/pinecone-io/pinecone-ts-client/blob/e683dafa91f67cb356cee61afddbb559bdc77ef7/src/pinecone-generated-ts-fetch/models/UpdateRequest.ts#L101

The setMetadata is supposed to be reassigned to set_metadata according to the pinecone API.

**Comment by NCKLS** [comment]
@rschwabco 

I've created a PR that resolves this as well as a sparseValues problem: https://github.com/pinecone-io/pinecone-ts-client/pull/40

Hope this is of use!

**Comment by realdylanwebb** [comment]
I am having this issue as well

**Comment by ashburnham** [comment]
It doesn't work for me either..

```
case 5: return [2 /*return*/, Promise.reject(new Error("PineconeClient: Error calling ".concat(func.name.replace("bound ", ""), ": ").concat(error)))];
                                                             ^

[Error: PineconeClient: Error calling upsert: TypeError: stream.getReader is not a function]
```

**Comment by NCKLS** [comment]
You can copy my mr in your local node modules to resolve it until the contributors consider merging it: https://github.com/pinecone-io/pinecone-ts-client/pull/40

**Comment by rschwabco** [comment]
@NCKLS The `update` currently function returns an empty object (that might change in the next major release of the client). 

Using the latest version of the client, I was able to update a vector as follows:

First, I insert the vector:
```ts
const vector: Vector = {
  id: "test",
  values: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10],
}

const upsertRequest: UpsertRequest = {
  vectors: [vector],
  namespace: "my-namespace",
}

const upsertResponse = await index.upsert({ upsertRequest })
```
Which results in:

```ts
{ upsertedCount: 1 }
```

Then I update the vector:

```ts
const updateRequest: UpdateRequest = {
  id: "test",
  values: [10, 2, 3, 4, 5, 6, 7, 8, 9, 10],
  setMetadata: {
    "test": "test",
  },
  namespace: "my-namespace",
}

const updateResponse = await index.update({ updateRequest })
console.log({ updateResponse })
```

This results in an empty object, as expected:
```ts
{ updateResponse: {} }
```

To see the updated, I fetch the vector by ID:

```ts
const fetchedVector: FetchResponse = await index.fetch({ ids: ["test"], namespace: "my-namespace" })
console.log(JSON.stringify(fetchedVector, null, 2))
```

You can see the vector has been updated with the new values as well as the new metadata:
```json
{
  "vectors": {
    "test": {
      "id": "test",
      "values": [
        10,
        2,
        3,
        4,
        5,
        6,
        7,
        8,
        9,
        10
      ],
      "metadata": {
        "test": "test"
      }
    }
  },
  "namespace": "my-namespace"
}
```
I've confirmed that sparse values are being update as well.

**Comment by rschwabco** [comment]
@ashburnham - Could you provide more information and context to reproduce the issue you're seeing?

**Comment by jhamon** [comment]
This is a pretty old issue and we've made a large number of changes since it was filed so I'm not sure it's still relevant. Feel free to file a new issue if you're still having problems with this.

Last week we shipped a big v1 release with a ton of improvements. Check out the [release notes](https://github.com/pinecone-io/pinecone-ts-client/releases) and [v1 migration guide](https://github.com/pinecone-io/pinecone-ts-client/blob/main/v1-migration.md) to get started using the new client.
